contracts/contracts/vault/VaultCore.sol

```diff
@@ -23,11 +23,8 @@ import "./VaultInitializer.sol";
 contract VaultCore is VaultInitializer {
     using SafeERC20 for IERC20;
     using StableMath for uint256;
-    // max signed int
-    uint256 internal constant MAX_INT = 2**255 - 1;
-    // max un-signed int
-    uint256 internal constant MAX_UINT =
-        0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+    /// @dev max signed int
+    uint256 internal constant MAX_INT = type(uint256).max;
```

Comment:
This needs to be max signed int, not unsigned int as the `abs` function does
`require(x < int256(MAX_INT), "Amount too high");`

This change breaks mainnet so I'll revert. No need to redeploy Sonic as `mintForStrategy` is not used. Although it will be when an AMO is added.

MAX_UINT can still be deleted